### PR TITLE
clean up some unnecessary files

### PR DESCRIPTION
**Pull request by guihao-liang**
* .flake8.log we don't need to track it in our repo.
* black doesn't need to check py38.
* ignore .pre-commit-config.yaml so that one can have their own customized pre-commit config living together with the repo.

### CODE REVIEW

**Comment by guihao-liang** [comment]
intentionally ignore,

* E501, line too long
* E266 too many leading '#' for block comment
* temporarily ignore all __init__.py files

```bash
flake8 src/python/turicreate --config=tox.ini | wc -l
   864
```

File: tox.ini [hunk 1]
```diff
@@ -1,3 +1,4 @@
 [flake8]
 max-line-length = 88
-extend-ignore = E501
+extend-ignore = E501, E203, E501, E266
```
Comment (TobyRoseman) [inline tox.ini]: I don't think you need to list `E501` twice.

File: .gitignore [hunk 2]
```diff
@@ -105,3 +105,6 @@ deps/build
 # CCache within Docker
 /.docker_ccache/
 /.docker_images/
+
+# Pre-commit stuff
```
Comment (TobyRoseman) [inline .gitignore]: I don't really think this comment is really adding anything. I'd probably prefer just not having a comment here than having this comment. A better comment might be `# git hooks`.

Comment (guihao-liang) [reply .gitignore]: That's better! Will do.

File: tox.ini (hunk 1 above)
Comment (guihao-liang) [reply tox.ini]: Yeah, my oversight.

**Comment by guihao-liang** [comment]
> Do we want to add instructions for running flake8 to our [build document](https://github.com/apple/turicreate/blob/master/BUILD.md)?
> 
> Do we want to add `flake8` to our [development requirements.txt](https://github.com/apple/turicreate/blob/master/scripts/requirements.txt)?

No, this flake8 is only used by pre-commit hooks and also personal use. Currently, there are too many things that we need to fix in order to use flake8.

**Pull request merged**